src/main/java/foxtrot/uniform/charlie/kilo/controller/OrderController.java

```diff
@@ -0,0 +1,47 @@
+package foxtrot.uniform.charlie.kilo.controller;
+
+import foxtrot.uniform.charlie.kilo.model.Order;
+import foxtrot.uniform.charlie.kilo.model.dto.OrderResponseDto;
+import foxtrot.uniform.charlie.kilo.service.OrderService;
+import foxtrot.uniform.charlie.kilo.service.ShoppingCartService;
+import foxtrot.uniform.charlie.kilo.service.UserService;
+import foxtrot.uniform.charlie.kilo.service.implementation.dto.MapperToDto;
+import foxtrot.uniform.charlie.kilo.service.implementation.dto.OrderResponseMapperToDto;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/orders")
+public class OrderController {
+    private final OrderService orderService;
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
+    private final MapperToDto<OrderResponseDto, Order> responseMapper;
+
+    @Autowired
+    public OrderController(OrderService orderService, UserService userService,
+                           ShoppingCartService shoppingCartService,
+                           OrderResponseMapperToDto responseMapper) {
```

Comment:
```suggestion
                           MapperToDto<OrderResponseDto, Order> responseMapper) {
```
we should not depend on the implementation